src/main/java/com/traveltime/sdk/TravelTimeSDK.java

```diff
@@ -77,14 +77,24 @@ private <T> Either<TravelTimeError, T> parseJsonBody(TravelTimeRequest<T> reques
     }
 
     private <T> Either<TravelTimeError, T> deserializeProtoResponse(ProtoRequest<T> request, Response response) {
-        Either<TravelTimeError, T> protoResponse = Try
-            .of(() -> Objects.requireNonNull(response.body()).bytes())
-            .toEither()
-            .<TravelTimeError>mapLeft(cause -> new IOError(cause, IO_CONNECTION_ERROR + cause.getMessage()))
-            .flatMap(request::parseBytes);
-
-        response.close();
-        return protoResponse;
+        String url = response.request().url().toString();
+        switch(response.code()){
+            case 404: {
+                NetworkError error = new NetworkError("Network response is 404 (Not found). " + url
```

Comment:
`404` is not a network error but a request error. That is, the sender of the request misconfigured it somehow.

The message `does not exist, has been moved, or has a broken link.` is improper:
 - `does not exist` doesn't tell you anything more than `not found`
 - `has been moved` would be a `3XX` code, probably `301` or `302`
 - `has a broken link` doesn't make much sense, but if the URI is malformed then we wouldn't even get a response

Instead say something like "make sure the URL $url is correct"

Reply:
Alright, I just googled 404 definition and made the message according to that :) The damn google